### PR TITLE
Split unit tests and integration tests, and update workflows

### PR DESCRIPTION
**Pull request by Jacqueline-J**
This pull request addresses issue #263 "Split integration tests and unit tests"

-  Tests in the Satip/tests folder were split into unit and integration tests
-  Replaced pytest with pytest tests/unitTest and pytest tests/integrationTest in the workflows.yaml file
-  test_utils.py remained unorganized as it's currently disabled

### CODE REVIEW

File: tests/integration_test/download_integration_test.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"""integration Tests for satip.download.py."""
```
Comment (peterdudfield) [inline tests/integration_test/download_integration_test.py:1]: For pytest, I think the file name needs to be `test_{extra name}.py` for it to run? But I could be wrong, its certainly common pratice to do this. 

This is probably the same with the rest of the tests

**Review by peterdudfield — CHANGES_REQUESTED** [review]
Thanks so much for doing this, I think its just one small change, to change the files names to have test at the start. 

Ill have a think how we can try to test this all works before merging.

**Comment by peterdudfield** [comment]
> Thanks so much for doing this, I think its just one small change, to change the files names to have test at the start.
> 
> Ill have a think how we can try to test this all works before merging.

Oh and then's one small fix in the `pre-commit.cr` error

**Comment by Jacqueline-J** [comment]
Hi Peter, 

Initially, I named the files using prefixes like "integration_test" or "unit_test." However, upon running checks, I noticed that no tests were being detected or executed.

After reviewing the “Conventions for Python test discovery”, I found that file names should either start with test_*.py or end with *_test.py. I changed the name of all the test files and the code passed the checks successfully. But, I understand the importance of adhering to established naming conventions and best practices, so I'm more than willing to make further adjustments by changing the names to start with "test".

**Comment by Jacqueline-J** [comment]
> > Thanks so much for doing this, I think its just one small change, to change the files names to have test at the start.
> > Ill have a think how we can try to test this all works before merging.
> 
> Oh and then's one small fix in the `pre-commit.cr` error

Took awhile but I found the issue, it should be fine now :)

**Comment by peterdudfield** [comment]
Thanks @Jacqueline-J , I'm gona merge this to a branch internal first. Just so we can make sure the integration tests work. 

One small things I do that, becasue the coverage will only look at integration tests, I wonder could 
1. Try to merge coverage reports, using something like this - https://pytest-cov.readthedocs.io/en/latest/readme.html#coverage-data-file
2. run both unit and integration tests second, then the coverage would have all the tests in it

**Comment by Jacqueline-J** [comment]
@peterdudfield sorry for the oversight! I've appended the code to combine the coverage reports

**Comment by peterdudfield** [comment]
> @peterdudfield sorry for the oversight! I've appended the code to combine the coverage reports

No problem, thanks for doing that. Let me know when your happy and i'll merge (perhaps into a different branch) to check it works all ok

**Review by peterdudfield — APPROVED** [review]

**Comment by peterdudfield** [comment]
Looks ready to go, shall I merge it into `development` and then we can see the unit and integration tests work?

Thanks again for all this work

**Comment by Jacqueline-J** [comment]
Sorry for the slight delay! Everything should be good now. Please go ahead and merge it.

**Pull request merged**